new-nextjs-client/src/components/index-page/AddYearDropdown/AddYearDropdown.tsx

```diff
@@ -24,11 +26,33 @@ const getYearOptions = (): YearOption[] => {
 
 function AddYearDropdown() {
   const { appUser, updateAppUser } = useAppUser();
+  const [yearOptions, setYearOptions] = useState(getYearOptions());
+
+  const showToastBox = useAppToast();
 
   const onSelectChange = (newValue: YearOption) => {
-    updateAppUser((draft) => {
-      draft.addYear(newValue.value);
-    });
+    if (!appUser.years.includes(newValue.value)) {
+      updateAppUser((draft) => {
+        draft.addYear(newValue.value);
+        draft.sortYear();
+
+        showToastBox({
+          status: 'success',
+          highlightedData: newValue.label,
+          message: 'year added',
+          duration: 3500,
+        });
+      });
+
+      const newYearOptionsArr = yearOptions.map((obj) => {
```

Comment:
Let's make a separate helper function for this. you can name `disableSelectedOptions()` then call `setYearOptions()` inside it with the updated options.